src/win32/kerberos_win32.cc

```diff
@@ -12,182 +26,139 @@
 #define GSS_C_REPLAY_FLAG 4
 #define GSS_C_SEQUENCE_FLAG 8
 
-const std::wstring to_wstring(const char *bytes) {
-    DWORD sizeOfStr = MultiByteToWideChar(CP_UTF8, 0, bytes, -1, NULL, 0);
-    assert(sizeOfStr > 0);
-    std::wstring arg(sizeOfStr, '\0');
-    DWORD result = MultiByteToWideChar(CP_UTF8, 0, bytes, -1, &arg[0], sizeOfStr);
-    assert(result > 0);
-    arg.resize(result - 1);
-    return arg;
-}
-
-NAN_INLINE std::wstring WStringOptionValue(v8::Local<v8::Object> options, const char* _key) {
-    Nan::HandleScope scope;
-    v8::Local<v8::String> key = Nan::New(_key).ToLocalChecked();
-    if (options.IsEmpty() || !Nan::Has(options, key).FromMaybe(false)) {
-      return std::wstring();
-    }
-
-    v8::Local<v8::Value> value = Nan::Get(options, key).ToLocalChecked();
-    if (!value->IsString()) {
-      return std::wstring();
-    }
-
-    return to_wstring(*(Nan::Utf8String(value)));
-}
-
 /// KerberosClient
-KerberosClient::~KerberosClient() {
-    if (_state != NULL) {
-        auth_sspi_client_clean(_state);
-        free(_state);
-        _state = NULL;
-    }
-}
-
-NAN_METHOD(KerberosClient::Step) {
-    KerberosClient* client = Nan::ObjectWrap::Unwrap<KerberosClient>(info.This());
-    std::string challenge(*Nan::Utf8String(info[0]));
-    Nan::Callback* callback = new Nan::Callback(Nan::To<v8::Function>(info[1]).ToLocalChecked());
+void KerberosClient::Step(const CallbackInfo& info) {
+    auto state = this->state();
```

Comment:
I think if you disagree with my earlier comment about returning a raw pointer, then we can at least `const auto&` here?

Reply:
The point is that this _should_ be a copy – if you think using `std::shared_ptr<>` instead of `auto` here would help with making that clear, I’m also happy to switch that :)

Reply:
Nope, sorry I didn't understand why we wanted the copy above, resolving.